nixos/modules/config/iproute2.nix

```diff
@@ -18,15 +26,14 @@ in
   };
 
   config = mkIf cfg.enable {
-    environment.etc."iproute2/bpf_pinning" = { mode = "0644"; text = fileContents "${pkgs.iproute2}/etc/iproute2/bpf_pinning"; };
-    environment.etc."iproute2/ematch_map"  = { mode = "0644"; text = fileContents "${pkgs.iproute2}/etc/iproute2/ematch_map";  };
-    environment.etc."iproute2/group"       = { mode = "0644"; text = fileContents "${pkgs.iproute2}/etc/iproute2/group";       };
-    environment.etc."iproute2/nl_protos"   = { mode = "0644"; text = fileContents "${pkgs.iproute2}/etc/iproute2/nl_protos";   };
-    environment.etc."iproute2/rt_dsfield"  = { mode = "0644"; text = fileContents "${pkgs.iproute2}/etc/iproute2/rt_dsfield";  };
-    environment.etc."iproute2/rt_protos"   = { mode = "0644"; text = fileContents "${pkgs.iproute2}/etc/iproute2/rt_protos";   };
-    environment.etc."iproute2/rt_realms"   = { mode = "0644"; text = fileContents "${pkgs.iproute2}/etc/iproute2/rt_realms";   };
-    environment.etc."iproute2/rt_scopes"   = { mode = "0644"; text = fileContents "${pkgs.iproute2}/etc/iproute2/rt_scopes";   };
-    environment.etc."iproute2/rt_tables"   = { mode = "0644"; text = (fileContents "${pkgs.iproute2}/etc/iproute2/rt_tables")
-                                                                   + (optionalString (cfg.rttablesExtraConfig != "") "\n\n${cfg.rttablesExtraConfig}"); };
+    environment.etc."iproute2/bpf_pinning" = { mode = "0644"; source = "${iproute_etc}/bpf_pinning"; };
```

Comment:
Can we use

```nix
   environment.etc."iproute2" = { mode = "0644"; source = "${iproute_etc}/*"; };
```

here?

Reply:
I don't think so, no. From my testing
```nix
    environment.etc."iproute2/rt_tables"   = { mode = "0644"; source = rt_tables;                    };
    environment.etc."iproute2"   = { mode = "0644"; source = "${iproute_etc}/*";                     };
```
result in a collision where (at least in my case) the rt_tables from the iproute2 package will win over the custom rt_tables. 

Tho in theory it would also be possible to create a derivation for `/etc/iproute2` linking to everything but rt_tables.

But what we could do, is:

```nix

let
  etc_iproute2 = pkgs.stdenv.mkDerivation {
    name = "etc-iproute2";
    src = pkgs.iproute2;
    buildPhase = ''
      mkdir $out
      ln -s $src/etc/iproute2/* $out/
      rm -f $out/rt_tables
      echo ${lib.escapeShellArg cfg.rttablesExtraConfig} | cat $src/etc/iproute2/rt_tables - > $out/rt_tables
    '';
  };
in {
  # [...]
  config.environment.etc."iproute2".source = mkIf cfg.enable etc_iproute2;
};
```

I personally like that approach, because we won't need manual adjustments in the nix code when new files in `/etc/iproute2` appear.

Reply:
Yes, the glob links the whole directory so we can't just link another file into it.

Reply:
Yes, and if we create derivation containing the entire directory anyways I personally see no need to link link all single files `iproute2/*` instead of the entire directory.

Reply:
I've implemented the mentioned directory derivation. I think that should address your proposal.